### ISSUE TITLE
Event update:country type question

### ISSUE THREAD

**Issue by ydedev** [paul-thebaud/v-phone-input#13]
## Expected Behavior

get <html>
<body>

CountryIso2
--

<br class="Apple-interchange-newline">
</body>
</html>

## Actual Behavior

get string

## Steps to Reproduce the Problem

  1. The event update:country did not get the CountryIso2 type, but actually got the string type

```
<script setup>
import { ref } from 'vue';
import {
	getCountryCodeForRegionCode,
} from 'awesome-phonenumber'

const phone = ref('');
let phoneRef = ref()
function submit(){
    console.log(phone)
    console.log(phoneRef.value)
}


function countryFunc(e){
    console.log(e)
    console.log(getCountryCodeForRegionCode( e ))
}

</script>

<template>

  <div style="margin: 0 auto; width: 430px;">
    <v-phone-input ref="phoneRef" v-model="phone"  @update:country="countryFunc"/>
    <v-btn @click="submit" block class="mt-2">Submit</v-btn>
  </div>

</template>
```

<html>
<body>
<h3 tabindex="-1" dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Events</h3>

Name | Type | Description
-- | -- | --
update:modelValue | string | Emitted when the country or phone is updated with the E164 formatted phone number.
update:country | CountryIso2 | Emitted when the country is updated with the selected country.

</body>
</html>

## Specifications

  - OS:win11
  - Browser:chrome
`    "v-phone-input": "^3.0.5",
    "vue": "^3.3.4",
    "vue-router": "^4.2.2",
    "vuetify": "^3.3.5",`
Events
Name	Type	Description
update:modelValue	string	Emitted when the country or phone is updated with the E164 formatted phone number.
update:country	[CountryIso2](https://github.com/paul-thebaud/v-phone-input#country)	Emitted when the country is updated with the selected country.

**Comment by paul-thebaud** [comment]
Hello,
Sorry, but I don't understand your question. Could you rephrase in a simple maner it please?

**Comment by ydedev** [comment]
In the readme description, the update:country event type value is CountryIso2, and the actual value obtained is a string.

**Comment by paul-thebaud** [comment]
OK. `CountryIso2` is a type alias just representing a string of 2 chars.
See the following [Typescript types](https://github.com/paul-thebaud/v-phone-input/blob/8cf427c0e743f21ef2f9d8340932c081b042827b/src/types/countries.d.ts#L1).

**Comment by procodingtools** [comment]
Sorry but this is a stupid thing. why it doesn't emitting the entire country object?

**Comment by paul-thebaud** [comment]
@procodingtools,
Because country can be bidirectionally bound (using `v-model`), and that would not be convenient to use an object instead of a string value.
In addition, most apps are using the input to manage a scalar value (the phone number or country string), and, for example, exchanging it with an API. Emitting an object instead of a scalar value would make those most common use cases more verbose.

**Comment by paul-thebaud** [comment]
Another thing @procodingtools: please be respectful. Instead of saying "this is a stupid thing", prefer using "I don't think it it should work this way". Thanks.
